pkg/lib/synthesize/resource_getters.go

```diff
@@ -0,0 +1,240 @@
+	"github.com/operator-framework/operator-registry/pkg/registry"
+)
+
+const (
+	CSVKind                = "ClusterServiceVersion"
+	CRDKind                = "CustomResourceDefinition"
+	SecretKind             = "Secret"
+	ConfigMapKind          = "ConfigMap"
+	ClusterRoleKind        = "ClusterRole"
+	ClusterRoleBindingKind = "ClusterRoleBinding"
+	ServiceAccountKind     = "ServiceAccount"
+	ServiceKind            = "Service"
+	RoleKind               = "Role"
+	RoleBindingKind        = "RoleBinding"
+	PrometheusRuleKind     = "PrometheusRule"
+	ServiceMonitorKind     = "ServiceMonitor"
+	DeploymentKind         = "Deployment"
+)
+
+// registerGVK registers the GVK of the kube resource we use in the CSV, explicitly including all versions.
+func registerGVK(scheme *runtime.Scheme) {
+	scheme.AddKnownTypes(operatorsv1alpha1.SchemeGroupVersion, &operatorsv1alpha1.ClusterServiceVersion{})
+
+	scheme.AddKnownTypes(apiextensionsv1.SchemeGroupVersion, &apiextensionsv1.CustomResourceDefinition{})
+	scheme.AddKnownTypes(apiextensionsv1beta1.SchemeGroupVersion, &apiextensionsv1beta1.CustomResourceDefinition{})
+
+	scheme.AddKnownTypes(corev1.SchemeGroupVersion,
+		&corev1.ServiceAccount{},
+		&corev1.ConfigMap{},
+		&corev1.Secret{})
+
+	scheme.AddKnownTypes(rbacv1.SchemeGroupVersion,
+		&rbacv1.Role{},
+		&rbacv1.RoleBinding{},
+		&rbacv1.ClusterRole{},
+		&rbacv1.ClusterRoleBinding{})
+
+	scheme.AddKnownTypes(appsv1.SchemeGroupVersion, &appsv1.Deployment{})
+}
+
+func NewGVKN(gvk schema.GroupVersionKind, name string) registry.DefinitionKey {
+	return registry.DefinitionKey{
+		Group:   gvk.Group,
+		Version: gvk.Version,
+		Kind:    gvk.Kind,
+		Name:    name,
+	}
+}
+
+type customResourceDefinition struct {
+	v1      apiextensionsv1.CustomResourceDefinition
+	v1beta1 apiextensionsv1beta1.CustomResourceDefinition
+	gvkn    registry.DefinitionKey
+}
+
+func (crd *customResourceDefinition) getResourceGVKNs() (gvkns []registry.DefinitionKey) {
+	switch crd.gvkn.Version {
+	case crd.v1.GroupVersionKind().Version:
+		for _, version := range crd.v1.Spec.Versions {
+			gvkns = append(gvkns, registry.DefinitionKey{
+				Group:   crd.v1.Spec.Group,
+				Kind:    crd.v1.Spec.Names.Kind,
+				Version: version.Name,
+				Name:    crd.v1.GetName(),
+			})
+		}
+	case crd.v1beta1.GroupVersionKind().Version:
+		verions := make(map[string]struct{})
+		for _, ver := range crd.v1beta1.Spec.Versions {
+			verions[ver.Name] = struct{}{}
+		}
+		// Although version is deprecated, this checks if it is still used.
+		verions[crd.v1beta1.Spec.Version] = struct{}{}
+
+		for ver := range verions {
+			if ver == "" {
+				continue
+			}
+
+			gvkns = append(gvkns, registry.DefinitionKey{
+				Group:   crd.v1beta1.Spec.Group,
+				Kind:    crd.v1beta1.Spec.Names.Kind,
+				Version: ver,
+				Name:    crd.v1beta1.GetName(),
+			})
+		}
+	}
+	return
+}
+
+func (s *synthesize) getCRDs() ([]customResourceDefinition, error) {
+	if crdIFs, ok := s.unstructuredObjectMap[CRDKind]; ok {
```

Comment:
Nit: It can be nice for readability to invert the boolean expression and return early, to align the happy path along the left hand side. But in this case, checking `ok` is unnecessary -- you can range over a nil map!